pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,701 @@
+
+        Returns
+        -------
+        bool
+            ``True`` if Auto/Manual SMatrix only solve is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.AutoSMatOnlySolve
+
+    @property
+    def enforce_causality(self):
+        """Whether to enforce causality during interpolating sweep.
+
+        Returns
+        -------
+        bool
+            ``True`` if enforce causality is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.EnforceCausality
+
+    @property
+    def enforce_dc_and_causality(self):
+        """Whether to enforce DC point and causality.
+
+        Returns
+        -------
+        bool
+            ``True`` if enforce dc point and causality is used, ``False`` otherwise.
+
+        """
+        return self._edb_sweep_data.EnforceDCAndCausality
+
+    @property
+    def enforce_passivity(self):
+        """Whether to enforce passivity during interpolating sweep.
+
+        Returns
+        -------
+        bool
+            ``True`` if enforce passivity is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.EnforcePassivity
+
+    @property
+    def freq_sweep_type(self):
+        """Sweep type.
+        Options are:
+        - ``kInterpolatingSweep``.
+        - ``kDiscreteSweep``.
+        - ``kBroadbandFastSweep``.
+
+        Returns
+        -------
+        str
+        """
+        return self._edb_sweep_data.FreqSweepType.ToString()
+
+    @property
+    def interp_use_full_basis(self):
+        """Whether to use Full basis elements.
+
+        Returns
+        -------
+        bool
+            ``True`` if full basis interpolation is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.InterpUseFullBasis
+
+    @property
+    def interp_use_port_impedance(self):
+        """Whether to turn on the port impedance interpolation.
+
+        Returns
+        -------
+        bool
+            ``True`` if port impedance is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.InterpUsePortImpedance
+
+    @property
+    def interp_use_prop_const(self):
+        """Whether to use propagation constants.
+
+        Returns
+        -------
+        bool
+            ``True`` if propagation constants are used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.InterpUsePropConst
+
+    @property
+    def interp_use_s_matrix(self):
+        """Whether to use S matrix.
+
+        Returns
+        -------
+        bool
+            ``True`` if S matrix are used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.InterpUseSMatrix
+
+    @property
+    def max_solutions(self):
+        """Number of aximum solutions.
+
+        Returns
+        -------
+        int
+        """
+        return self._edb_sweep_data.MaxSolutions
+
+    @property
+    def min_freq_s_mat_only_solve(self):
+        """Minimum frequency SMatrix only solve.
```

Comment:
Can this be rewritten in plain English? Maybe "Frequency with units from the S matrix-only solve."